Build fails when Docker auth check fails, rather than proceeding with a warning

We encountered a [bug in testcontainers](https://github.com/testcontainers/testcontainers-java/issues/9055) that exposed an issue in `DockerService.getAuthConfigFor`. The testcontainers bug causes the incorrect URL to be set in the auth config, and thus the remote call to check the auth fails.

However, instead of getting a non-success `AuthResponse`, an `UnauthorizedException` is thrown (see stack trace), and the build fails completely, rather than proceeding with a warning, which seems to be the intended behavior. I believe that the push would ultimately succeed in spite of this, because the credentials would be combined with the image’s (correct) server URL—this worked correctly with v3 of the plugin, and only broke when we upgraded to v4.

This is reproducible using `docker-credential-gcloud` to push to a gcr registry.

```
[ERROR] Failed to execute goal io.micronaut.maven:micronaut-maven-plugin:4.6.1:docker-push (default-docker-push) on project: Status 401: {"message":"Get \"https://registry-1.docker.io/v2/\": unauthorized: incorrect username or password"} -> [Help 1]
 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.micronaut.maven:micronaut-maven-plugin:4.6.1:docker-push (default-docker-push) on project: Status 401: {"message":"Get \"https://registry-1.docker.io/v2/\": unauthorized: incorrect username or password"}
 
     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:903)
     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:280)
     at org.apache.maven.cli.MavenCli.main (MavenCli.java:203)
     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
     at java.lang.reflect.Method.invoke (Method.java:580)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
     at java.lang.reflect.Method.invoke (Method.java:580)
     at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
     at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
     at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
 Caused by: org.apache.maven.plugin.MojoExecutionException: Status 401: {"message":"Get \"https://registry-1.docker.io/v2/\": unauthorized: incorrect username or password"}
 
     at io.micronaut.maven.DockerPushMojo.execute (DockerPushMojo.java:91)
     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:903)
     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:280)
     at org.apache.maven.cli.MavenCli.main (MavenCli.java:203)
     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
     at java.lang.reflect.Method.invoke (Method.java:580)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
     at java.lang.reflect.Method.invoke (Method.java:580)
     at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
     at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
     at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
 Caused by: com.github.dockerjava.api.exception.UnauthorizedException: Status 401: {"message":"Get \"https://registry-1.docker.io/v2/\": unauthorized: incorrect username or password"}
 
     at com.github.dockerjava.core.DefaultInvocationBuilder.execute (DefaultInvocationBuilder.java:239)
     at com.github.dockerjava.core.DefaultInvocationBuilder.post (DefaultInvocationBuilder.java:124)
     at com.github.dockerjava.core.exec.AuthCmdExec.execute (AuthCmdExec.java:26)
     at com.github.dockerjava.core.exec.AuthCmdExec.execute (AuthCmdExec.java:13)
     at com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec (AbstrSyncDockerCmdExec.java:21)
     at com.github.dockerjava.core.command.AbstrDockerCmd.exec (AbstrDockerCmd.java:33)
     at com.github.dockerjava.core.command.AuthCmdImpl.exec (AuthCmdImpl.java:24)
     at io.micronaut.maven.services.DockerService.getAuthConfigFor (DockerService.java:289)
     at io.micronaut.maven.DockerPushMojo.execute (DockerPushMojo.java:83)
     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
     at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
     at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
     at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:903)
     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:280)
     at org.apache.maven.cli.MavenCli.main (MavenCli.java:203)
     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
     at java.lang.reflect.Method.invoke (Method.java:580)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
     at java.lang.reflect.Method.invoke (Method.java:580)
     at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
     at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
     at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
```

--- a/micronaut-maven-integration-tests/src/it/issue1172-deploy/invoker.properties
+++ b/micronaut-maven-integration-tests/src/it/issue1172-deploy/invoker.properties
@@ -1,0 +1,3 @@
+invoker.goals.1 = -ntp deploy -Dpackaging=docker -Djib.to.image=gcr.io/oraclelabs/micronaut-maven-plugin:0.1 -Djib.httpTimeout=0
+invoker.os.family = !windows
+invoker.buildResult = failure

--- a/micronaut-maven-integration-tests/src/it/issue1172-deploy/pom.xml
+++ b/micronaut-maven-integration-tests/src/it/issue1172-deploy/pom.xml
@@ -1,0 +1,119 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>io.micronaut.build.examples</groupId>
+  <artifactId>issue1172-deploy</artifactId>
+  <version>0.1</version>
+  <packaging>${packaging}</packaging>
+
+  <parent>
+    <groupId>io.micronaut.platform</groupId>
+    <artifactId>micronaut-parent</artifactId>
+    <version>@it.micronaut.version@</version>
+  </parent>
+
+  <properties>
+    <micronaut.version>@it.micronaut.version@</micronaut.version>
+    <micronaut-maven-plugin.version>@project.version@</micronaut-maven-plugin.version>
+    <exec.mainClass>io.micronaut.build.examples.Application</exec.mainClass>
+    <maven.native.plugin.version>@native-maven-plugin.version@</maven.native.plugin.version>
+    <jib-maven-plugin.version>@jib-maven-plugin.version@</jib-maven-plugin.version>
+    <packaging>jar</packaging>
+    <micronaut.runtime>netty</micronaut.runtime>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.micronaut</groupId>
+      <artifactId>micronaut-http-server-netty</artifactId>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.micronaut.serde</groupId>
+      <artifactId>micronaut-serde-jackson</artifactId>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.micronaut</groupId>
+      <artifactId>micronaut-http-client</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.micronaut.test</groupId>
+      <artifactId>micronaut-test-junit5</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.micronaut.maven</groupId>
+        <artifactId>micronaut-maven-plugin</artifactId>
+        <version>${micronaut-maven-plugin.version}</version>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <!-- Uncomment to enable incremental compilation -->
+          <!-- <useIncrementalCompilation>false</useIncrementalCompilation> -->
+          <annotationProcessorPaths combine.children="append">
+            <path>
+              <groupId>io.micronaut</groupId>
+              <artifactId>micronaut-http-validation</artifactId>
+              <version>${micronaut.core.version}</version>
+            </path>
+            <path>
+              <groupId>io.micronaut.serde</groupId>
+              <artifactId>micronaut-serde-processor</artifactId>
+              <version>${micronaut.serialization.version}</version>
+            </path>
+          </annotationProcessorPaths>
+          <compilerArgs>
+            <arg>-Amicronaut.processing.group=io.micronaut.build.examples</arg>
+            <arg>-Amicronaut.processing.module=${project.artifactId}</arg>
+          </compilerArgs>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>jib-maven-plugin</artifactId>
+        <configuration>
+          <to>
+            <auth>
+              <username>not-a-valid-user</username>
+              <password>not-a-valid-password</password>
+            </auth>
+          </to>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+</project>

--- a/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/main/java/io/micronaut/build/examples/Application.java
+++ b/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/main/java/io/micronaut/build/examples/Application.java
@@ -1,0 +1,10 @@
+package io.micronaut.build.examples;
+
+import io.micronaut.runtime.Micronaut;
+
+public class Application {
+
+    public static void main(String[] args) {
+        Micronaut.run(Application.class);
+    }
+}

--- a/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/main/resources/application.yml
+++ b/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/main/resources/application.yml
@@ -1,0 +1,3 @@
+micronaut:
+  application:
+    name: issue1172-deploy

--- a/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/main/resources/logback.xml
+++ b/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/main/resources/logback.xml
@@ -1,0 +1,15 @@
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <withJansi>false</withJansi>
+        <!-- encoders are assigned the type
+             ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
+        <encoder>
+            <pattern>%cyan(%d{HH:mm:ss.SSS}) %gray([%thread]) %highlight(%-5level) %magenta(%logger{36}) - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="info">
+        <appender-ref ref="STDOUT" />
+    </root>
+</configuration>

--- a/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/test/java/io/micronaut/build/examples/ApplicationTest.java
+++ b/micronaut-maven-integration-tests/src/it/issue1172-deploy/src/test/java/io/micronaut/build/examples/ApplicationTest.java
@@ -1,0 +1,21 @@
+package io.micronaut.build.examples;
+
+import io.micronaut.runtime.EmbeddedApplication;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import jakarta.inject.Inject;
+
+@MicronautTest
+public class ApplicationTest {
+
+    @Inject
+    EmbeddedApplication application;
+
+    @Test
+    void testItWorks() {
+        Assertions.assertTrue(application.isRunning());
+    }
+
+}

--- a/micronaut-maven-integration-tests/src/it/issue1172-deploy/verify.groovy
+++ b/micronaut-maven-integration-tests/src/it/issue1172-deploy/verify.groovy
@@ -1,0 +1,5 @@
+File log = new File(basedir, 'build.log')
+assert log.exists()
+assert log.text.contains("Built image to Docker daemon") : "Image should always be built"
+assert log.text.contains("[WARNING] Failed to login to registry") : "Credentials check should be a soft failure"
+assert log.text.contains("Pushing image:") : "A push should always be attempted if credentials are configured"

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/Dockerfile
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/Dockerfile
@@ -1,0 +1,7 @@
+# use the GCR mirror registry to ensure authentication failure
+FROM mirror.gcr.io/eclipse-temurin:17-jre
+WORKDIR /home/app
+COPY classes /home/app/classes
+COPY dependency/* /home/app/libs/
+EXPOSE 8080
+ENTRYPOINT ["java", "-cp", "/home/app/libs/*:/home/app/classes/", "io.micronaut.build.examples.Application"]

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/invoker.properties
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/invoker.properties
@@ -1,0 +1,3 @@
+invoker.goals.1 = -ntp package -Dpackaging=docker -Djib.httpTimeout=0
+invoker.os.family = !windows
+invoker.buildResult = failure

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/pom.xml
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/pom.xml
@@ -1,0 +1,121 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>io.micronaut.build.examples</groupId>
+  <artifactId>issue1172-package</artifactId>
+  <version>0.1</version>
+  <packaging>${packaging}</packaging>
+
+  <parent>
+    <groupId>io.micronaut.platform</groupId>
+    <artifactId>micronaut-parent</artifactId>
+    <version>@it.micronaut.version@</version>
+  </parent>
+
+  <properties>
+    <micronaut.version>@it.micronaut.version@</micronaut.version>
+    <micronaut-maven-plugin.version>@project.version@</micronaut-maven-plugin.version>
+    <exec.mainClass>io.micronaut.build.examples.Application</exec.mainClass>
+    <maven.native.plugin.version>@native-maven-plugin.version@</maven.native.plugin.version>
+    <jib-maven-plugin.version>@jib-maven-plugin.version@</jib-maven-plugin.version>
+    <packaging>jar</packaging>
+    <micronaut.runtime>netty</micronaut.runtime>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.micronaut</groupId>
+      <artifactId>micronaut-http-server-netty</artifactId>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.micronaut.serde</groupId>
+      <artifactId>micronaut-serde-jackson</artifactId>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.micronaut</groupId>
+      <artifactId>micronaut-http-client</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.micronaut.test</groupId>
+      <artifactId>micronaut-test-junit5</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>io.micronaut.maven</groupId>
+        <artifactId>micronaut-maven-plugin</artifactId>
+        <version>${micronaut-maven-plugin.version}</version>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <!-- Uncomment to enable incremental compilation -->
+          <!-- <useIncrementalCompilation>false</useIncrementalCompilation> -->
+          <annotationProcessorPaths combine.children="append">
+            <path>
+              <groupId>io.micronaut</groupId>
+              <artifactId>micronaut-http-validation</artifactId>
+              <version>${micronaut.core.version}</version>
+            </path>
+            <path>
+              <groupId>io.micronaut.serde</groupId>
+              <artifactId>micronaut-serde-processor</artifactId>
+              <version>${micronaut.serialization.version}</version>
+            </path>
+          </annotationProcessorPaths>
+          <compilerArgs>
+            <arg>-Amicronaut.processing.group=io.micronaut.build.examples</arg>
+            <arg>-Amicronaut.processing.module=${project.artifactId}</arg>
+          </compilerArgs>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>jib-maven-plugin</artifactId>
+        <configuration>
+          <from>
+            <!-- use the GCR mirror registry to ensure authentication failure -->
+            <image>mirror.gcr.io/eclipse-temurin:17-jre</image>
+            <auth>
+              <username>not-a-valid-user</username>
+              <password>not-a-valid-password</password>
+            </auth>
+          </from>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+</project>

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/src/main/java/io/micronaut/build/examples/Application.java
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/src/main/java/io/micronaut/build/examples/Application.java
@@ -1,0 +1,10 @@
+package io.micronaut.build.examples;
+
+import io.micronaut.runtime.Micronaut;
+
+public class Application {
+
+    public static void main(String[] args) {
+        Micronaut.run(Application.class);
+    }
+}

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/src/main/resources/application.yml
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/src/main/resources/application.yml
@@ -1,0 +1,3 @@
+micronaut:
+  application:
+    name: issue1172-package

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/src/main/resources/logback.xml
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/src/main/resources/logback.xml
@@ -1,0 +1,15 @@
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <withJansi>false</withJansi>
+        <!-- encoders are assigned the type
+             ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
+        <encoder>
+            <pattern>%cyan(%d{HH:mm:ss.SSS}) %gray([%thread]) %highlight(%-5level) %magenta(%logger{36}) - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="info">
+        <appender-ref ref="STDOUT" />
+    </root>
+</configuration>

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/src/test/java/io/micronaut/build/examples/ApplicationTest.java
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/src/test/java/io/micronaut/build/examples/ApplicationTest.java
@@ -1,0 +1,21 @@
+package io.micronaut.build.examples;
+
+import io.micronaut.runtime.EmbeddedApplication;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import jakarta.inject.Inject;
+
+@MicronautTest
+public class ApplicationTest {
+
+    @Inject
+    EmbeddedApplication application;
+
+    @Test
+    void testItWorks() {
+        Assertions.assertTrue(application.isRunning());
+    }
+
+}

--- a/micronaut-maven-integration-tests/src/it/issue1172-package/verify.groovy
+++ b/micronaut-maven-integration-tests/src/it/issue1172-package/verify.groovy
@@ -1,0 +1,4 @@
+File log = new File(basedir, 'build.log')
+assert log.exists()
+assert log.text.contains("[WARNING] Failed to login to registry") : "Credentials check should be a soft failure"
+assert log.text.contains("Could not build image") : "Build should be attempted"

--- a/micronaut-maven-plugin/src/main/java/io/micronaut/maven/services/DockerService.java
+++ b/micronaut-maven-plugin/src/main/java/io/micronaut/maven/services/DockerService.java
@@ -287,8 +287,17 @@ public class DockerService {
             .withPassword(password);
         RegistryAuthLocator registryAuthLocator = RegistryAuthLocator.instance();
         AuthConfig authConfig = registryAuthLocator.lookupAuthConfig(dockerImageName, defaultAuthConfig);
-        AuthResponse authResponse = dockerClient.authCmd().withAuthConfig(authConfig).exec();
-        if (authResponse.getStatus() != null && authResponse.getStatus().equals("Login Succeeded")) {
+        boolean loginSucceeded = false;
+        try {
+            AuthResponse authResponse = dockerClient.authCmd().withAuthConfig(authConfig).exec();
+            if (authResponse.getStatus() != null && authResponse.getStatus().equals("Login Succeeded")) {
+                loginSucceeded = true;
+            }
+        } catch (Exception ignored) {
+            // typically this is com.github.dockerjava.api.exception.UnauthorizedException
+        }
+
+        if (loginSucceeded) {
             LOG.info("Successfully logged in to registry {}", dockerImageName.getRegistry());
         } else {
             LOG.warn("Failed to login to registry {}", dockerImageName.getRegistry());

